cranelift/codegen/build.rs

```diff
@@ -56,9 +62,17 @@ fn main() {
         crate_dir.join("build.rs").to_str().unwrap()
     );
 
-    if let Err(err) = meta::generate(&isas, &out_dir) {
-        eprintln!("Error: {}", err);
-        process::exit(1);
+    if let Some(new_backend_isas) = new_backend_isas {
+        debug_assert!(isas.is_empty());
+        if let Err(err) = meta::generate_new_backend(&new_backend_isas, &out_dir) {
+            eprintln!("Error: {}", err);
+            process::exit(1);
+        }
+    } else {
+        if let Err(err) = meta::generate(&isas, &out_dir) {
+            eprintln!("Error: {}", err);
+            process::exit(1);
+        }
```

Comment:
Shorter is: `let result = if let Some(...` and do the `if let Err(err) ...` check afterwards.